tests/CMakeLists.txt

```diff
@@ -6,8 +6,21 @@
 # v. 2.0. If a copy of the MPL was not distributed with this file, You can
 # obtain one at http://mozilla.org/MPL/2.0/.
 
-# All the tests here will require Google Test, so we will go ahead and find it.
-find_package(GTest CONFIG REQUIRED)
+# All the tests here will require Google Test, so we will go ahead and set it
+# up.
+# Set up GoogleTest.
+option(COVFIE_SETUP_GOOGLETEST "Set up the GoogleTest targets explicitly"
+    TRUE)
+option(COVFIE_USE_SYSTEM_GOOGLETEST
+    "Pick up an existing installation of GoogleTest from the build environment"
+    TRUE)
+if(COVFIE_SETUP_GOOGLETEST)
+    if(COVFIE_USE_SYSTEM_GOOGLETEST)
+        find_package(GTest CONFIG REQUIRED)
+    else()
+        add_subdirectory(googletest)
+    endif()
+endif()
```

Comment:
Is there any good reason for any project importing this to ever need to build the tests? I'm okay with the `COVFIE_USE_SYSTEM_GOOGLETEST` flag but I'm not sure if the `COVFIE_SETUP_GOOGLETEST` flag would ever be useful?

Reply:
It is indeed not too likely that we'll want to build the covfie unit tests when building covfie as part of a downstream project. But on the off chance that we ever want to do that, it's good to have the theoretical ability.

The logic with the two cache variables is the same that we use in all the other R&D projects. So I'm keen to use it in the same way here as well.